### ISSUE TITLE
dataclasses package interferes with python versions > 3.7 ... sometimes?

### ISSUE THREAD

**Issue by biblicabeebli** [onnela-lab/forest#129]
Somewhere between commits e72ac1d4fd36698a050867218fd5894c5ea16e9d and 2402c6f73d3e562dc346e39ba0408cd9f4607857 a requirements update added the `dataclasses` library. (it is up to date, 0.6)

The error does not occur in all contexts, but it does occur when running alongside the Django dependencies (possibly only when Django loaded? its a complex framework) inside an `ipython` shell on the beiwe-backend data processing servers, as well as my development environment.

I can manually uninstall the dataclasses package (everything works, it seems to be genuinely unnecessary in 3.7+) for manual debugging, but it will keep getting installed whenever forest is updated, or when pip runs the requirements_data_processing.py file during automated operations.

I tracked down another library that had the same problem, and the modification made to the setup.py file that fixed the issue:
https://github.com/google/flax/pull/270/commits/00f7fd7374fcd315525663369f999deb242cbfea

While `dataclasses;python_version<'3.7'` is syntactically valid to stick in a requirements.txt file it does not resolve the issue when I specify it in the requirements_data_processing.txt file in the beiwe-backend codebase.

@hackdna do you have any idea what could be going on here?  I have never seen something quite this obscure.
Would you mind if we test to see whether adding that secret invocation in the setup.py file fixes the issue?

The monstrous stacktrace is as follows, you can also trigger the latter half with just `from pandas.io.common import *` in the same context:
```
In [2]: from forest.jasmine.traj2stats import gps_stats_main
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 from forest.jasmine.traj2stats import gps_stats_main

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/forest/jasmine/traj2stats.py:13, in <module>
     10 from typing import Dict, List, Tuple, Union
     12 import numpy as np
---> 13 import pandas as pd
     14 from pyproj import Transformer
     15 import requests

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/__init__.py:48, in <module>
     45 # let init-time option registration happen
     46 import pandas.core.config_init
---> 48 from pandas.core.api import (
     49     # dtype
     50     Int8Dtype,
     51     Int16Dtype,
     52     Int32Dtype,
     53     Int64Dtype,
     54     UInt8Dtype,
     55     UInt16Dtype,
     56     UInt32Dtype,
     57     UInt64Dtype,
     58     Float32Dtype,
     59     Float64Dtype,
     60     CategoricalDtype,
     61     PeriodDtype,
     62     IntervalDtype,
     63     DatetimeTZDtype,
     64     StringDtype,
     65     BooleanDtype,
     66     # missing
     67     NA,
     68     isna,
     69     isnull,
     70     notna,
     71     notnull,
     72     # indexes
     73     Index,
     74     CategoricalIndex,
     75     RangeIndex,
     76     MultiIndex,
     77     IntervalIndex,
     78     TimedeltaIndex,
     79     DatetimeIndex,
     80     PeriodIndex,
     81     IndexSlice,
     82     # tseries
     83     NaT,
     84     Period,
     85     period_range,
     86     Timedelta,
     87     timedelta_range,
     88     Timestamp,
     89     date_range,
     90     bdate_range,
     91     Interval,
     92     interval_range,
     93     DateOffset,
     94     # conversion
     95     to_numeric,
     96     to_datetime,
     97     to_timedelta,
     98     # misc
     99     Flags,
    100     Grouper,
    101     factorize,
    102     unique,
    103     value_counts,
    104     NamedAgg,
    105     array,
    106     Categorical,
    107     set_eng_float_format,
    108     Series,
    109     DataFrame,
    110 )
    112 from pandas.core.arrays.sparse import SparseDtype
    114 from pandas.tseries.api import infer_freq

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/core/api.py:48, in <module>
     46 from pandas.core.construction import array
     47 from pandas.core.flags import Flags
---> 48 from pandas.core.groupby import (
     49     Grouper,
     50     NamedAgg,
     51 )
     52 from pandas.core.indexes.api import (
     53     CategoricalIndex,
     54     DatetimeIndex,
   (...)
     64     UInt64Index,
     65 )
     66 from pandas.core.indexes.datetimes import (
     67     bdate_range,
     68     date_range,
     69 )

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/core/groupby/__init__.py:1, in <module>
----> 1 from pandas.core.groupby.generic import (
      2     DataFrameGroupBy,
      3     NamedAgg,
      4     SeriesGroupBy,
      5 )
      6 from pandas.core.groupby.groupby import GroupBy
      7 from pandas.core.groupby.grouper import Grouper

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/core/groupby/generic.py:70, in <module>
     68 import pandas.core.common as com
     69 from pandas.core.construction import create_series_with_explicit_dtype
---> 70 from pandas.core.frame import DataFrame
     71 from pandas.core.generic import NDFrame
     72 from pandas.core.groupby import base

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/core/frame.py:157, in <module>
    151 from pandas.core.arrays.sparse import SparseFrameAccessor
    152 from pandas.core.construction import (
    153     extract_array,
    154     sanitize_array,
    155     sanitize_masked_array,
    156 )
--> 157 from pandas.core.generic import NDFrame
    158 from pandas.core.indexers import check_key_length
    159 from pandas.core.indexes.api import (
    160     DatetimeIndex,
    161     Index,
   (...)
    165     ensure_index_from_sequences,
    166 )

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/core/generic.py:129, in <module>
    124 import pandas.core.common as com
    125 from pandas.core.construction import (
    126     create_series_with_explicit_dtype,
    127     extract_array,
    128 )
--> 129 from pandas.core.describe import describe_ndframe
    130 from pandas.core.flags import Flags
    131 from pandas.core.indexes.api import (
    132     DatetimeIndex,
    133     Index,
   (...)
    138     ensure_index,
    139 )

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/core/describe.py:38, in <module>
     29 from pandas.core.dtypes.common import (
     30     is_bool_dtype,
     31     is_datetime64_any_dtype,
     32     is_numeric_dtype,
     33     is_timedelta64_dtype,
     34 )
     36 from pandas.core.reshape.concat import concat
---> 38 from pandas.io.formats.format import format_percentiles
     40 if TYPE_CHECKING:
     41     from pandas import (
     42         DataFrame,
     43         Series,
     44     )

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/io/formats/format.py:99, in <module>
     96 from pandas.core.indexes.timedeltas import TimedeltaIndex
     97 from pandas.core.reshape.concat import concat
---> 99 from pandas.io.common import (
    100     check_parent_directory,
    101     stringify_path,
    102 )
    103 from pandas.io.formats.printing import (
    104     adjoin,
    105     justify,
    106     pprint_thing,
    107 )
    109 if TYPE_CHECKING:

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/pandas/io/common.py:69, in <module>
     63 _RFC_3986_PATTERN = re.compile(r"^[A-Za-z][A-Za-z0-9+\-+.]*://")
     65 BaseBufferT = TypeVar("BaseBufferT", bound=BaseBuffer)
     68 @dataclasses.dataclass
---> 69 class IOArgs:
     70     """
     71     Return value of io/common.py:_get_filepath_or_buffer.
     72     """
     74     filepath_or_buffer: str | BaseBuffer

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/dataclasses.py:958, in dataclass(_cls, init, repr, eq, order, unsafe_hash, frozen)
    955     return wrap
    957 # We're called as @dataclass without parens.
--> 958 return wrap(_cls)

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/dataclasses.py:950, in dataclass.<locals>.wrap(cls)
    949 def wrap(cls):
--> 950     return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/dataclasses.py:800, in _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
    795 cls_annotations = cls.__dict__.get('__annotations__', {})
    797 # Now find fields in our class.  While doing so, validate some
    798 # things, and set the default values (as class attributes) where
    799 # we can.
--> 800 cls_fields = [_get_field(cls, name, type)
    801               for name, type in cls_annotations.items()]
    802 for f in cls_fields:
    803     fields[f.name] = f

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/dataclasses.py:800, in <listcomp>(.0)
    795 cls_annotations = cls.__dict__.get('__annotations__', {})
    797 # Now find fields in our class.  While doing so, validate some
    798 # things, and set the default values (as class attributes) where
    799 # we can.
--> 800 cls_fields = [_get_field(cls, name, type)
    801               for name, type in cls_annotations.items()]
    802 for f in cls_fields:
    803     fields[f.name] = f

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/dataclasses.py:659, in _get_field(cls, a_name, a_type)
    657 typing = sys.modules.get('typing')
    658 if typing:
--> 659     if (_is_classvar(a_type, typing)
    660         or (isinstance(f.type, str)
    661             and _is_type(f.type, cls, typing, typing.ClassVar,
    662                          _is_classvar))):
    663         f._field_type = _FIELD_CLASSVAR
    665 # If the type is InitVar, or if it's a matching string annotation,
    666 # then it's an InitVar.

File ~/.pyenv/versions/3.8.13/envs/scrb/lib/python3.8/site-packages/dataclasses.py:550, in _is_classvar(a_type, typing)
    547 def _is_classvar(a_type, typing):
    548     # This test uses a typing internal class, but it's the best way to
    549     # test if this is a ClassVar.
--> 550     return type(a_type) is typing._ClassVar

AttributeError: module 'typing' has no attribute '_ClassVar'

```

**Comment by hackdna** [comment]
This library is a backport for Python 3.6 and should have been removed when Forest switched to 3.8. I think we can safely get rid of it now.

**Comment by hackdna** [comment]
Fixed in c4e2661
